src/misc/support.cpp

```diff
@@ -331,3 +331,15 @@ std::string safe_strerror(int err) noexcept
 	return buf;
 #endif
 }
+
+void set_thread_name(std::thread& thread, const char *name)
+{
+#if HAVE_PTHREAD_SETNAME_NP
+	pthread_t handle = thread.native_handle();
+	pthread_setname_np(handle, name);
```

Comment:
You should add an assert before calling `pthread_setname_np`:
```C++
assert(strlen(name) < 16);
```

Reply:
Great suggestion; thanks @dreamer.